data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/GenesisTimeHandlerTest.java

```diff
@@ -13,30 +13,37 @@
 
 package tech.pegasys.artemis.beaconrestapi.beaconhandlers;
 
-import static org.assertj.core.api.Assertions.assertThat;
+import static javax.servlet.http.HttpServletResponse.SC_INTERNAL_SERVER_ERROR;
 
 import com.google.common.eventbus.EventBus;
 import com.google.common.primitives.UnsignedLong;
+import io.javalin.http.Context;
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
-import tech.pegasys.artemis.beaconrestapi.handlerinterfaces.BeaconRestApiHandler.RequestParams;
+import tech.pegasys.artemis.provider.JsonProvider;
 import tech.pegasys.artemis.storage.ChainStorageClient;
 
-class GenesisTimeHandlerTest {
-  private final RequestParams requestParams = Mockito.mock(RequestParams.class);
+public class GenesisTimeHandlerTest {
+  private Context mockContext = Mockito.mock(Context.class);
+  private final UnsignedLong genesisTime = UnsignedLong.valueOf(51234);
+
   private final ChainStorageClient storageClient =
       ChainStorageClient.memoryOnlyClient(new EventBus());
-  private final GenesisTimeHandler handler = new GenesisTimeHandler(storageClient);
 
   @Test
-  public void shouldReturnEmptyObjectWhenGenesisTimeIsNotSet() {
-    assertThat(handler.handleRequest(requestParams)).isEqualTo(null);
+  public void shouldRaiseInternalErrorWhenGenesisTimeIsNotSet() throws Exception {
```

Comment:
I'm not sure this should be an internal server error.  There is a, potentially long, period where a node is running prior to the genesis event occurring and we should have some meaningful response during that time (it may still be an error, but an expected one, not an internal server error).  404 Not found is probably the most accurate in REST terms but likely to be confusing for users who may think they got the URL wrong.

During that period we would have a `storageClient` but it's `store` and `genesisTime` fields would be `null`.